### PR TITLE
Add waitForService to TS integration test

### PR DESCRIPTION
**Pull request by RyanRHall**
Fixes race condition in TS integration test by waiting for CL node before initiating test suit.

Waiting for https://github.com/smartcontractkit/chainlink/pull/2476

### CODE REVIEW

File: tools/docker/compose [hunk 1]
```diff
@@ -47,8 +47,8 @@ Commands:
     dev:integration | di  Run docker-compose with dev config for integration tests
     dev                   Run docker-compose with dev config for the core node and operator-ui
 
-    eth:clean             Reset blockchain data to genesis state
-    cl:clean              Reset chainlink database
+    eth:restart             Reset blockchain data to genesis state
+    cl:restart              Reset chainlink database
```
Comment (RyanRHall) [inline tools/docker/compose]: unrelated to this PR - fixes `compose help` outpus

File: tools/ci-ts/test-helpers/common.ts [hunk 2]
```diff
@@ -103,10 +103,22 @@ export async function deployContract<T extends DeployFactory>(
 
 export async function wait(ms: number) {
   return new Promise(res => {
-    setTimeout(() => res(), ms)
+    setTimeout(res, ms)
   })
 }
 
+export async function waitForService(endpoint: string, timeout = 20000) {
```
Comment (HenryNguyen5) [inline tools/ci-ts/test-helpers/common.ts]: Can we use jsdoc here? e.g what the endpoint format is, what the timeout duration resolution is

File: tools/ci-ts/test-helpers/common.ts [hunk 3]
```diff
@@ -103,10 +103,22 @@ export async function deployContract<T extends DeployFactory>(
...
+  await assertAsync(
+    async () => {
+      return fetch(endpoint)
+        .then(() => true)
+        .catch(() => false)
+    },
```
Comment (HenryNguyen5) [inline tools/ci-ts/test-helpers/common.ts]:
```suggestion
  await assertAsync(
    () => 
       fetch(endpoint)
        .then(() => true)
        .catch(() => false)
```

File: tools/ci-ts/test-helpers/common.ts [hunk 4]
```diff
@@ -103,10 +103,27 @@ export async function deployContract<T extends DeployFactory>(
...
-    setTimeout(() => res(), ms)
+    setTimeout(res, ms)
   })
 }
 
+/**
+ * Makes a simple get request to an endpoint and ensures the service responds.
+ * Status code doesn't matter - just ensures the service is running.
+ * @param endpoint the url of the service
+ * @param timeout the time to wait before erroring
```
Comment (HenryNguyen5) [inline tools/ci-ts/test-helpers/common.ts]:
```suggestion
 * @param timeout the time in milliseconds to wait before erroring
```

**Comment by RyanRHall** [comment]
Going to close this PR and merge these changes into the PR for the multi-node integration test

**Pull request closed without merging**